daft/runners/ray_runner.py

```diff
@@ -135,11 +174,54 @@ def _make_ray_block_from_micropartition(partition: MicroPartition) -> RayDataset
         return partition.to_pylist()
 
 
+def _series_from_arrow_with_ray_data_extensions(
+    array: pa.Array | pa.ChunkedArray, name: str = "arrow_series"
+) -> Series:
+    if _RAY_DATA_EXTENSIONS_AVAILABLE and isinstance(array.type, ArrowTensorType):
+        storage_series = _series_from_arrow_with_ray_data_extensions(array.storage, name=name)
+        series = storage_series.cast(
+            DataType.fixed_size_list(
+                _from_arrow_type_with_ray_data_extensions(array.type.scalar_type),
+                int(np.prod(array.type.shape)),
+            )
+        )
+        return series.cast(DataType.from_arrow_type(array.type))
+    elif _RAY_DATA_EXTENSIONS_AVAILABLE and isinstance(array.type, ArrowVariableShapedTensorType):
+        return Series.from_numpy(array.to_numpy(zero_copy_only=False), name=name)
+    else:
+        return Series.from_arrow(array, name)
+
+
+def _micropartition_from_arrow_with_ray_data_extensions(arrow_table: pa.Table) -> MicroPartition:
+    assert isinstance(arrow_table, pa.Table)
+    non_native_fields = [
+        arrow_field.name
+        for arrow_field in arrow_table.schema
+        if (dt := _from_arrow_type_with_ray_data_extensions(arrow_field)) == DataType.python()
+        or dt._is_tensor_type()
+        or dt._is_fixed_shape_tensor_type()
+    ]
+    if non_native_fields:
```

Comment:
Any downside to just always using `MicroPartition._from_tables`? Is this a perf optimization?

Reply:
Calling `MicroPartition.from_arrow` in the other path is strictly for abstraction (and deduplication) purposes, and in fact might have a negligible perf hit, because it goes through this same schema check.

The alternative is bringing in record batch logic here, which isn't too bad, it would just be:
```
arrow_table = ensure_table(arrow_table)
pyt = _PyTable.from_arrow_record_batches(arrow_table.to_batches(), schema._schema)
return Table._from_pytable(pyt)
```

Reply:
Discussed offline, keeping this for performance reasons